File: queue.js
class Queue {
  
    constructor() {
        this.items = [];
    }

    // Enqueue: Add an element to the rear of the queue
    enqueue(element) {
        this.items.push(element);
    }

    // Dequeue: Remove and return the element from the front of the queue
    dequeue() {
        if (this.isEmpty()) {
            return "Queue is empty";
        }
        return this.items.shift();
    }

    // Peek: Return the element at the front of the queue without removing it
    peek() {
        if (this.isEmpty()) {
            return "Queue is empty";
        }
        return this.items[0];
    }

    // isEmpty: Check if the queue is empty
    isEmpty() {
        return this.items.length === 0;
    }

    // size: Return the number of elements in the queue
    size() {
        return this.items.length;
    }
}

// Example usage:
let queue = new Queue();

queue.enqueue(1);
 // create a object // push element one by one  
queue.enqueue(2);
queue.enqueue(3);

console.log("Front element:", queue.peek()); // Output: Front element: 1
console.log("Dequeued element:", queue.dequeue()); // Output: Dequeued element: 1

console.log("Queue size:", queue.size()); // Output: Queue size: 2

File: queue.py
# Actually, deque (pronounced "deck") stands for double-ended queue,
# which is a data structure that allows insertion and deletion of elements from both the front and the rear of the queue.
# It's different from the FIFO (First-In-First-Out) queue,
# where elements are removed in the order they were added. 
# deque provides efficient O(1) time complexity for insertion and deletion operations at both ends of the queue.


# __init__: Initializes an empty queue.
# self.values = []: Checks if the queue is empty.
# enqueue: Adds an item to the end of the queue.
# dequeue: Removes and returns the item at the front of the queue.
# peek: Returns the item at the front of the queue without removing it.
# size: Returns the number of items in the queue.


class queue:
def __init__(self):
self.values = []
def enqueue(self, x):
self.values.append(x)
def dequeue(self):
front = self.values[0]
self.values= self.values[1:]
return front
q1 = queue(10)
q1.enqueue(20)
q1.enqueue(30)
print(q1.values)
q1.enqueue(40)
print(q1.values) 

output = [10, 20,30]
[10,20,30,40]
